### PR TITLE
Add notification docs

### PR DESCRIPTION
**Pull request by taysea**
Adds missing notification:

- [x] `notification.direction`
- [x] `notification.global.direction`
- [x] `notification.global.container`
- [x] `notification.toast.direction`
- [x] `notification.critical.background`
- [x] `notification.critical.global.background`
- [x] `notification.critical.toast.background`
- [x] `notification.warning.background`
- [x] `notification.warning.global.background`
- [x] `notification.warning.toast.background`
- [x] `notification.normal.background`
- [x] `notification.normal.color`
- [x] `notification.normal.global.background`
- [x] `notification.warning.toast.background`
- [x] `notification.info.icon`
- [x] `notification.info.background`
- [x] `notification.info.color`
- [x] `notification.info.global.background`
- [x] `notification.info.toast.background`
- [x] `notification.unknown.background`
- [x] `notification.unknown.global.background`
- [x] `notification.unknown.toast.background`
- [x] `notification.undefined.background`
- [x] `notification.undefined.global.background`
- [x] `notification.undefined.toast.background`

Closes https://github.com/grommet/grommet-site/issues/379

### CODE REVIEW

File: src/screens/Notification.js [hunk 1]
```diff
@@ -99,6 +99,27 @@ return (
...
+            any anchor properties.
+          </Description>
+          <PropertyValue type="array">
+            <Example>{`[{ label: 'Renew Subscription', href: '/renew'}]`}</Example>
+            <Example>{`[{ label: 'Renew Subscription', href: '/renew'}, { label: 'View Details', href:'/details' }]`}</Example>
+          </PropertyValue>
+        </Property>
+        <Property name="global">
+          <Description>
+            A boolean flag that indicates whether or not the Notification is a
```
Comment (halocline) [inline src/screens/Notification.js:115]:
```suggestion
            A boolean flag indicating whether or not the Notification is a
```

File: src/screens/Notification.js [hunk 2]
```diff
@@ -193,6 +214,58 @@ return (
...
+    horizontal: 'large',
+    vertical: 'xsmall',
+  },
+                `}
+            </Example>
+          </PropertyValue>
+        </Property>
+        <Property name="notification.direction">
+          <Description>
+            The layout direction of the title and message.
```
Comment (halocline) [inline src/screens/Notification.js:235]:
```
The layout direction for the title and message.
```

File: src/screens/Notification.js [hunk 3]
```diff
@@ -193,6 +214,58 @@ return (
...
+            <Example defaultValue>
+              {`
+  {
+  pad: {
+    horizontal: 'large',
+    vertical: 'xsmall',
+  },
+                `}
+            </Example>
+          </PropertyValue>
```
Comment (halocline) [inline src/screens/Notification.js:247]: Is this correct? Was expecting 'row' and 'column'

File: src/screens/Notification.js [hunk 4]
```diff
@@ -193,6 +214,58 @@ return (
...
+        </Property>
+        <Property name="notification.direction">
```
Comment (halocline) [inline src/screens/Notification.js:233]: keep all .globals grouped together? Move this up alphabetically?

File: src/screens/Notification.js [hunk 5]
```diff
@@ -99,6 +99,27 @@ return (
...
       <Properties>
+        <Property name="actions">
+          <Description>
+            Anchors that appear to the right of the title and message. Value
+            should be provided as an array of objects, where the object accepts
+            any anchor properties.
+          </Description>
+          <PropertyValue type="array">
+            <Example>{`[{ label: 'Renew Subscription', href: '/renew'}]`}</Example>
+            <Example>{`[{ label: 'Renew Subscription', href: '/renew'}, { label: 'View Details', href:'/details' }]`}</Example>
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/screens/Notification.js]: Done.

File: src/screens/Notification.js (hunk 2 above)
Comment (taysea) [reply src/screens/Notification.js]: Done.

File: src/screens/Notification.js (hunk 3 above)
Comment (taysea) [reply src/screens/Notification.js:244]: Fixed.

File: src/screens/Notification.js (hunk 4 above)
Comment (taysea) [reply src/screens/Notification.js]: Done, but curious about your thoughts of how the statuses are handled wrt `global` `toast` groupings.

File: src/screens/Notification.js [hunk 6]
```diff
@@ -193,6 +217,52 @@ return (
...
+          <Description>
+            The layout direction for the title and message of a global
+            notification.
+          </Description>
+          <PropertyValue type="string">
+            <Example defaultValue>"row"</Example>
+            <Example>"column"</Example>
+          </PropertyValue>
+        </Property>
+        <Property name="notification.toast.direction">
```
Comment (halocline) [inline src/screens/Notification.js:256]: Out of alphabetical order?

File: src/screens/Notification.js (hunk 4 above)
Comment (halocline) [reply src/screens/Notification.js]: Like do they feel overkill or redundant? Can we create some sort of shorthand which references the properties?

Something like this?

```
notification.critical

object: {
  background: // any background properties,
  color: // any color properties,
  icon: <StatusCriticalSmall /> // any icon component or svg,
  ...
}
```

**Pull request merged**